### PR TITLE
Add indexer and create simple indexer flow

### PR DESCRIPTION
**Pull request by georgeamccarthy**
app.py opens a flow which computes the embeddings for the short protein file (note this file has proteins all of the same length, currently proteins of different lengths are not supported).

proteins.json is created (and not tracked) which stores a mostly human readable collection of documents in the proteins DocumentArray.

.gitignore modified to include DS_store for Mac operating systems and proteins.json because this file will frequently change.

### CODE REVIEW

**Comment by georgeamccarthy** [comment]
@fissoreg not sure why my relative imports not working, can you take a look at it please? :)

File: backend/my_executors.py [hunk 1]
```diff
@@ -2,11 +2,12 @@
...
 from transformers import BertModel, BertTokenizer
 import re
-from typing import Sequence, List
+from typing import Sequence, List, Dict, Tuple
+import os
 
 from jina import Executor, requests
 from jina import Document, DocumentArray
-
+from jina.types.arrays.memmap import DocumentArrayMemmap
```
Comment (fissoreg) [inline backend/my_executors.py:10]: This is not used, right? We should remove it.

File: tests/test_executors.py [hunk 2]
```diff
@@ -5,9 +5,27 @@
...
 def test_ProtBERT_encoding():
     sequences = ["A E T C Z A O", "T E A B Z A O"]
     docs = DocumentArray((Document(content=seq) for seq in sequences))
     embedded_docs = ProtBertExecutor().encode(docs)
+
+# %%
+def test_Indexer_indexing_samelength():
+    proteins = DocumentArray(
+        from_csv(
+            fp=open("../data/samelength.csv"),
```
Comment (fissoreg) [inline tests/test_executors.py:21]: Tests are run from the parent directory, so this should be `fp=open("data/samelength.csv")`. This should fix the `build` badge (as tests in the cloud are failing for this reason, I believe).

**Review by fissoreg — CHANGES_REQUESTED** [review]

File: tests/test_executors.py [hunk 3]
```diff
@@ -5,9 +5,27 @@
...
+    proteins = DocumentArray(
+        from_csv(
+            fp=open("data/samelength.csv"),
+            field_resolver={"Protein sequences": "text"},
+        )
+    )
+
+    flow = Flow(port_expose=12345).add(uses=ProtBertExecutor).add(uses=MyIndexer)
+    with flow:
+        flow.index(proteins)
```
Comment (fissoreg) [inline tests/test_executors.py:28]: The function `test_Indexer_indexing_samelength` is supposed to test the indexer, not the entire flow. We could instantiate `MyIndexer` directly like for the `ProtBertExecutor` test function. To reuse some code, we could do something like:

```suggestion
sequences = ["A E T C Z A O", "T E A B Z A O"]

def encode_sequences(sequences):
    docs = DocumentArray((Document(content=seq) for seq in sequences))
    embedded_docs = ProtBertExecutor().encode(docs)
    
    return embedded_docs

def test_ProtBERT_encoding():
    embedded_docs = encode_sequences(sequences)

# %%
def test_Indexer_indexing_samelength():
    embedded_docs = encode_sequences(sequences)
    
    # indexing
    indexed_docs = MyIndexer().index(embedded_docs)
```

Comment (fissoreg) [reply tests/test_executors.py:28]: We might also add a third test to test the complete flow, but as this PR is about the indexer that could also wait.

Comment (georgeamccarthy) [reply tests/test_executors.py]: Good idea, however the MyIndexer requires self.workspace to be initialised as so index cannot be used as a static method. I'll raise an issue and we can implement a test in the future.

Comment (fissoreg) [reply tests/test_executors.py]: I agree, let's open an issue and move on. I'll merge this.

**Pull request merged**